src/profile-logic/profile-data.js

```diff
@@ -1027,6 +1027,53 @@ export function filterThreadToSearchString(
   );
 }
 
+export function filterThreadByTab(
+  thread: Thread,
+  relevantPages: Set<InnerWindowID>
+): Thread {
+  return timeCode('filterThreadByTab', () => {
+    if (relevantPages.size === 0) {
+      return thread;
+    }
+
+    const { frameTable, stackTable } = thread;
+
+    const frameMatchesFilterCache = new Map();
```

Comment:
It would probably be helpful to add an inline type here, I find the `Map` type hard to reason with when it's implicit like this, and I think it's easy to use wrong and have things slip in there, like `null` and `undefined`.